simpler_eda/numerical_eda.py

```diff
@@ -80,5 +74,146 @@ def numerical_plot(
+    assert isinstance(x_transform, bool), error_four
+
+    error_five = "TypeError: y_transform must be of type boolean."
+    assert isinstance(y_transform, bool), error_five
+
+    error_six = "TypeError: plot_type must be either 'scatter' or 'line'."
+    assert plot_type in ["scatter", "line"], error_six
+
+    error_seven = "TypeError: plot_width must be an integer."
+    assert isinstance(plot_width, int), error_seven
+
+    error_eight = "TypeError: plot_height must be an integer."
+    assert isinstance(plot_height, int), error_eight
+
+    error_nine = "TypeError: font_size must be a positive integer."
+    assert isinstance(font_size, int), error_nine
+
+    error_ten = "Wrong type!: color must be a string."
+    assert isinstance(color, str), error_ten
+
+    error_eleven = "Wrong type!: title must be a string."
+    assert isinstance(title, str), error_eleven
+
+    error_twelve = "Wrong type!: color_scheme must be a string."
+    assert isinstance(color_scheme, str), error_twelve
+
+    # create a copy of the dataframe so that original dataframe
+    #  remains unchanged
+    df = data.copy(deep=True)
+
+    # Ensure variable exists in the dataframe
+    assert xval in df.columns, "Variable xval not found in input dataframe."
+
+    assert yval in df.columns, "Variable yval not found in input dataframe."
+
+    # Ensure variable is numeric
+    error_msg_x = "Your x-variable needs to be numeric."
+    assert pd.api.types.is_numeric_dtype(df[xval]), error_msg_x
+
+    error_msg_y = "Your y-variable needs to be numeric."
+    assert pd.api.types.is_numeric_dtype(df[yval]), error_msg_y
+
+    # Toggle a log transformation on the x-axis
+    warn_one = "Can't have negative x values with np.log"
+    if x_transform:
+        if any(df[xval] < 0):
+            warnings.warn(warn_one)
+        df[xval] = np.log(df[xval])
+        xval = xval
+
+    # Toggle a log transformation on the y-axis
+    warn_two = "Can't have negative y values with np.log"
+    if y_transform:
+        if any(df[yval] < 0):
+            warnings.warn(warn_two)
+        df[yval] = np.log(df[yval])
+
+    # Update scale bounds of the plots
+    x_scale = alt.Scale(domain=(float(df[xval].min()), float(df[xval].max())))
+    y_scale = alt.Scale(domain=(float(df[yval].min()), float(df[yval].max())))
+
+    # Renaming column names, replacing underscores with space
+    # Replacing underscores to whitespace
+    df.columns = df.columns.str.replace("_", " ")
+    xval = str(xval).replace("_", " ")
+    yval = str(yval).replace("_", " ")
+    color = str(color).replace("_", " ")
+
+    # Plotting code for the function, code for either plot_type in ['line', 'scatter']
+    # Code for the scatter plot
+    if plot_type == "scatter":
+        if color in df.columns:
+            numerical_eda = (
+                alt.Chart(df, title=alt.TitleParams(text=title))
+                .mark_circle(size=30, opacity=0.8)
```

Comment:
Will size=30 a bit huge?

Reply:
Yes that is right have changed it to a smaller value